frame/header-mmr/src/lib.rs

```diff
@@ -62,108 +62,159 @@ mod tests;
 
 #[frame_support::pallet]
 pub mod pallet {
+	pub mod types {
+		/// The type use for indexing a node
+		pub type NodeIndex = u64;
+	}
+	pub use types::*;
+
 	// --- crates.io ---
 	#[cfg(feature = "std")]
 	use serde::Serialize;
-	// --- github.com ---
-	use mmr::{Error, MMRStore, Merge, Result as MMRResult, MMR};
-	// --- substrate ---
-	use frame_support::pallet_prelude::*;
+	// --- paritytech ---
+	use frame_support::{pallet_prelude::*, weights::Weight};
 	use frame_system::pallet_prelude::*;
 	use sp_runtime::{
 		generic::{DigestItem, OpaqueDigestItemId},
-		traits::{Hash, Header, SaturatedConversion},
+		traits::{Header, SaturatedConversion},
 	};
 	use sp_std::prelude::*;
 	// --- darwinia ---
+	use crate::{primitives::*, weights::WeightInfo};
 	use darwinia_header_mmr_rpc_runtime_api::{Proof, RuntimeDispatchInfo};
 	use darwinia_relay_primitives::MMR as MMRT;
 
-	pub const PARENT_MMR_ROOT_LOG_ID: [u8; 4] = *b"MMRR";
+	// ? Useless const
+	// ?
+	// ? commented by Xavier
+	/// The prefix of [`MerkleMountainRangeRootLog`]
+	pub const LOG_PREFIX: [u8; 4] = *b"MMRR";
```

Comment:
Need to add comment if renaming, indicating that this is the mmr root generated by leaves to parent header.